src/k8s/pkg/k8sd/database/clusterconfigs/cluster_configs.go

```diff
@@ -45,27 +34,54 @@ type ClusterConfigCertificates struct {
 	FrontProxyCAKey        string `yaml:"front-proxy-ca-key,omitempty"`
 }
 
-type ClusterConfigCluster struct {
-	CIDR string `yaml:"cidr,omitempty"`
+type Kubelet struct {
+	CloudProvider string `yaml:"cloud-provider,omitempty"`
+	ClusterDNS    string `yaml:"cluster-dns,omitempty"`
+	ClusterDomain string `yaml:"cluster-domain,omitempty"`
 }
 
-// ClusterConfig is the control plane configuration format of the k8s cluster.
-// ClusterConfig should attempt to use structured fields wherever possible.
-type ClusterConfig struct {
-	Cluster      ClusterConfigCluster      `yaml:"cluster"`
-	Certificates ClusterConfigCertificates `yaml:"certificates"`
-	Kubelet      ClusterConfigKubelet      `yaml:"kubelet"`
-	APIServer    ClusterConfigAPIServer    `yaml:"apiserver"`
+type APIServer struct {
+	SecurePort          int    `yaml:"secure-port,omitempty"`
+	AuthorizationMode   string `yaml:"authorization-mode,omitempty"`
+	ServiceAccountKey   string `yaml:"service-account-key,omitempty"`
+	Datastore           string `yaml:"datastore,omitempty"`
+	DatastoreURL        string `yaml:"datastore-url,omitempty"`
+	DatastoreCA         string `yaml:"datastore-ca,omitempty"`
+	DatastoreClientCert string `yaml:"datastore-client-crt,omitempty"`
+	DatastoreClientKey  string `yaml:"datastore-client-key,omitempty"`
+}
+
+func Default() ClusterConfig {
+	return ClusterConfig{
+		Cluster: Cluster{
+			CIDR: "10.1.0.0/16",
+		},
+		Kubelet: Kubelet{
+			ClusterDomain: "cluster.local",
+			ClusterDNS:    "10.152.183.10",
```

Comment:
not a good idea, let's leave the dns component handle this. this brings clusters with doa dns before coredns is enabled.